### PR TITLE
Fixed issue #17963: Numbers on text question does not detect blanks

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: Number() remove white space then check if not update string

### CODE REVIEW

File: assets/scripts/expressions/em_javascript.js [hunk 1]
```diff
@@ -1021,7 +1021,10 @@ function LEMval(alias)
                     return "";
                 }
 
-                var checkNumericRegex = new RegExp(/^(-)?[0-9]*(,|\.)[0-9]*$/);
+                var isNumericRegex = new RegExp(/^(-)?[0-9]*(\.)[0-9]*$/);
```
Comment (gabrieljenik) [inline assets/scripts/expressions/em_javascript.js:1024]: This is used more than once. Maybe we can put it on a function named like `getNumericValidatioRegex`?

Comment (gabrieljenik) [reply assets/scripts/expressions/em_javascript.js:1024]: Not sure if this is a must for merging... Will leave it up to you @Shnoulle

Comment (Shnoulle) [reply assets/scripts/expressions/em_javascript.js:1024]: > This is used more than once. Maybe we can put it on a function named like `getNumericValidatioRegex`?

I just fix here, not rewrite.

Comment (Shnoulle) [reply assets/scripts/expressions/em_javascript.js:1024]: > Not sure if this is a must for merging... Will leave it up to you @Shnoulle

?

**Comment by olleharstedt** [comment]
Thanks!

**Pull request merged**